File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/DiagnosticReport.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class DiagnosticReport : DomainResource, IResource
{
    public Base.CodeableConcept[]? Category { get; set; }
    public Base.CodeableConcept[]? ConclusionCode { get; set; }
    public string? Conclusion { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Base.ResourceReference[]? Specimen { get; set; }
    public string? EffectiveDateTime { get; set; }
    public Base.ResourceReference[]? ResultsInterpreter { get; set; }
    public required string Status { get; set; }
    public Base.ResourceReference[]? Result { get; set; }
    public object? Effective    
    {
        get
        {
            if (EffectiveDateTime is not null)
            {
                return EffectiveDateTime;
            }
    
            if (EffectivePeriod is not null)
            {
                return EffectivePeriod;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                EffectiveDateTime = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(Base.Period))
            {
                EffectivePeriod = (Base.Period)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public required Base.CodeableConcept Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? Issued { get; set; }
    public Base.Attachment[]? PresentedForm { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public Base.ResourceReference[]? ImagingStudy { get; set; }
    public DiagnosticReport_Media[]? Media { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public Base.ResourceReference[]? Performer { get; set; }
    public Base.Period? EffectivePeriod { get; set; }

    public class DiagnosticReport_Media : BackboneElement
    {
        public required Base.ResourceReference Link { get; set; }
        public string? Comment { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/DiagnosticReport.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Attachment
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class DiagnosticReportMedia(BackboneElement):
    link: Reference
    comment: Optional[str] = None

@dataclass(kw_only=True)
class DiagnosticReport(DomainResource):
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    conclusion_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    conclusion: Optional[str] = None
    encounter: Optional[Reference] = None
    specimen: Optional[List[Reference]] = field(default_factory=list)
    effective_date_time: Optional[str] = None
    results_interpreter: Optional[List[Reference]] = field(default_factory=list)
    status: str
    result: Optional[List[Reference]] = field(default_factory=list)
    code: CodeableConcept
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    issued: Optional[str] = None
    presented_form: Optional[List[Attachment]] = field(default_factory=list)
    based_on: Optional[List[Reference]] = field(default_factory=list)
    imaging_study: Optional[List[Reference]] = field(default_factory=list)
    media: Optional[List[DiagnosticReportMedia]] = field(default_factory=list)
    subject: Optional[Reference] = None
    performer: Optional[List[Reference]] = field(default_factory=list)
    effective_period: Optional[Period] = None